### ISSUE TITLE
[TypeError: undefined is not a constructor (evaluating 'new jsencrypt_1.JSEncrypt()')]

### ISSUE THREAD

**Issue by maxkomarychev** [travist/jsencrypt#235]
Hi @travist 

I am working on a react-native application which used to use `jsencrypt@3.0.0-rc.1` and it was working fine, however when I upgraded to latest `3.2.1` started seeing this error:

```
[TypeError: undefined is not a constructor (evaluating 'new jsencrypt_1.JSEncrypt()')]
```

I have tried using different ways to import the library but none of them seem to help: `import JSEncrypt from 'jsencrypt'` and `import { JSEncrypt } from 'jsencrypt'`

I noticed there have been considerable changes in webpack.config, any chance it can be reverted or improved?

I don't have a reproducible public project at the moment but I'll be happy to provide if this helps.

thanks!

**Comment by AtesDANIS** [comment]
hi that worked for me;

jsencrypt version: 3.2.1
`
import { JSEncrypt } from 'jsencrypt'

const encrypt = new JSEncrypt({default_key_size:4096})
encrypt.setPublicKey(pub_key)
var json = JSON.stringify({ account: 'cb01435781', password: 'aaaaaa' })
var encrypted = encrypt.encrypt(json);
console.log('encrypted', encrypted)
`

but extremely slow on react native :/ looking for a solution too

**Comment by shehanGuruge** [comment]
@AtesDANIS  did you get this kind of an error while using it in RN?

`Possible Unhandled Promise Rejection (id: 0):
TypeError: undefined is not a constructor (evaluating 'new _jsencrypt.JSEncrypt({
                              default_key_size: 4096
                            })')
`

**Comment by AtesDANIS** [comment]
yes i remind that. replaced library with react-native-rsa-native because slowness of jsencrypt on React Native

import { RSA } from 'react-native-rsa-native';

 RSA.generateKeys(2048) // set key size
        .then(keys => {
            setPrivateKey(keys.private); //do whatever you want with keys, i updated state
            setPublicKey(keys.public); //do whatever you want with keys, i updated state
        });


 RSA.encrypt(message, publicKey)
                    .then(_encodedMessage => {
                        //some logic
                    }).catch((errr) => { //some logic more});


 RSA.decrypt(message, privateKey)
            .then(decodedMessage => {
               //some logic
            }).catch((errr) => { //some logic more });

**Comment by shehanGuruge** [comment]
@AtesDANIS yeah the thing is react-native-rsa-native supports IOS 10+ and I need something that supports IOS 9+  :(

**Comment by AtesDANIS** [comment]
> @AtesDANIS did you get this kind of an error while using it in RN?
> 
> `Possible Unhandled Promise Rejection (id: 0): TypeError: undefined is not a constructor (evaluating 'new _jsencrypt.JSEncrypt({ default_key_size: 4096 })') `

today i tried it again but as i said before its very slow. when i tried first encountered same error message.
this is how i overcome it

`
npm install --save jsencrypt

`

`
import JSEncrypt from 'jsencrypt'

let encrypt =  new JSEncrypt({default_key_size: 2048})
console.log(encrypt.getPrivateKey())
console.log(encrypt.getPublicKey())
`
